conans/test/integration/environment/test_env.py

```diff
@@ -95,35 +95,33 @@ def build_requirements(self):
                 self.build_requires("mygtest/1.0", force_host_context=True)
 
             def build(self):
-                mybuild_cmd = "mycmake.bat" if platform.system() == "Windows" else "mycmake.sh"
-                self.run(mybuild_cmd, env="conanbuildenv")
-                mytest_cmd = "mygtest.bat" if platform.system() == "Windows" else "mygtest.sh"
-                self.run(mytest_cmd, env="conanrunenv")
+                self.run("mycmake.bat", env="conanbuildenv")
```

Comment:
But this test did work for ``platform.system() != Windows``, with an ``.sh`` script. Probably the ``build()`` can use ``self.settings.os`` to decide which one, but seems this test is testing less than it did before.

Reply:
Well, this test is/was a mess. Previously the choice of `.bat` `.sh` on the conan client code was based on `platform.system()` so according to that, the test, independently of the applied profiles, could work in any operating system, but that was wrong! As this PR now solves that issue and the generated "launchers" follow the build profile, not every check can be done in all systems.

Reply:
yes, agree, probably this test should be splitted in Windows and Linux specific ones. Not necessarily in this PR, good for me atm.